### PR TITLE
Newton polynomial evaluation

### PR DESCRIPTION
**Pull request by jtlap**

### CODE REVIEW

File: include/eve/module/real/polynomial/detail/newton_impl.hpp [hunk 1]
```diff
@@ -0,0 +1,125 @@
+#pragma once
```
Comment (jfalcou) [inline include/eve/module/real/polynomial/detail/newton_impl.hpp:1]: This file need a header block and remvoe the comments ...

File: include/eve/module/real/polynomial/function/numeric/generic/newton.hpp [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+                                        , numeric_type const &
+                                        , T0 xx, R1 const & rc
+                                               , R2 const & rn) noexcept
+  requires (compatible_values<T0, typename R1::value_type> && !simd_value<R1>
+            && compatible_values<T0, typename R2::value_type> && !simd_value<R2>)
+  {
+    return detail::newton_impl(numeric_type(), xx, rc, rn);
+  }
+
+//   //================================================================================================
```
Comment (jfalcou) [inline include/eve/module/real/polynomial/function/numeric/generic/newton.hpp:47]: remove

File: include/eve/module/real/polynomial/function/pedantic/generic/newton.hpp [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+                                        , pedantic_type const &
+                                        , T0 xx, R1 const & rc
+                                        , R2 const & rn) noexcept
+  requires (compatible_values<T0, typename R1::value_type> && (!simd_value<R1>)
+            && compatible_values<T0, typename R2::value_type> && (!simd_value<R2>))
+  {
+    return detail::newton_impl(pedantic_type(), xx, rc, rn);
+  }
+
+//   //================================================================================================
```
Comment (jfalcou) [inline include/eve/module/real/polynomial/function/pedantic/generic/newton.hpp:47]: remove

File: include/eve/module/real/polynomial/function/regular/generic/newton.hpp [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+  EVE_FORCEINLINE constexpr auto newton_(EVE_SUPPORTS(cpu_)
+                                        , T0 xx, R1 const & rc
+                                               , R2 const & rn) noexcept
+  requires (compatible_values<T0, typename R1::value_type> && (!simd_value<R1>)
+            && compatible_values<T0, typename R2::value_type> && (!simd_value<R2>))
+  {
+    return detail::newton_impl(regular_type(), xx, rc, rn);
+  }
+
+//   //================================================================================================
```
Comment (jfalcou) [inline include/eve/module/real/polynomial/function/regular/generic/newton.hpp:45]: remove

File: test/unit/module/real/polynomial/newton.cpp [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+{
+  using eve::newton;
+  using eve::fma;
+  using eve::pedantic;
+  using eve::numeric;
+  using eve::one;
+  using v_t = typename T::value_type;
+  for( auto a0 :a)
+  {
+//     {
```
Comment (jfalcou) [inline test/unit/module/real/polynomial/newton.cpp:128]: remove

**Review by jfalcou — CHANGES_REQUESTED** [review]

File: include/eve/module/real/polynomial/detail/newton_impl.hpp (hunk 1 above)
Comment (jtlap) [reply include/eve/module/real/polynomial/detail/newton_impl.hpp:1]: done

File: include/eve/module/real/polynomial/function/numeric/generic/newton.hpp (hunk 2 above)
Comment (jtlap) [reply include/eve/module/real/polynomial/function/numeric/generic/newton.hpp:47]: done

Comment (jfalcou) [reply include/eve/module/real/polynomial/function/numeric/generic/newton.hpp:47]: Nope this is still here ;)

File: include/eve/module/real/polynomial/function/pedantic/generic/newton.hpp (hunk 3 above)
Comment (jfalcou) [reply include/eve/module/real/polynomial/function/pedantic/generic/newton.hpp:47]: Same, still to remove

File: test/unit/module/real/polynomial/newton.cpp (hunk 5 above)
Comment (jfalcou) [reply test/unit/module/real/polynomial/newton.cpp:128]: Still to remove here too

**Comment by keryell** [comment]
I van varely unverstand vat is voing on in vis vommit...

**Review by jfalcou — APPROVED** [review]

**Pull request merged**